lib/services/elasticsearch.js

```diff
@@ -703,17 +703,90 @@ function refreshIndexIfNeeded(data, response) {
 }
 
 function formatESError(error) {
-  var kuzzleError = new Error(error.message);
+  var
+    formattedMessages,
+    message,
+    kuzzleError;
+
+  formattedMessages = {
+    // [illegal_argument_exception] object mapping [titi] can\'t be changed from nested to non-nested
+    '^\\[illegal_argument_exception\\] object mapping \\[([^\\]]+)\\] can\'t be changed from nested to non-nested$': 'Can not change mapping for field "$1" from nested to another type',
+
+    // [illegal_argument_exception] object mapping [baz] can\'t be changed from non-nested to nested
+    '^\\[illegal_argument_exception\\] object mapping \\[([^\\]]+)\\] can\'t be changed from non-nested to nested$': 'Can not change mapping for field "$1" from object to another type',
```

Comment:
Still, it can also occur when updating a scalar type to a nested one, in which case the original type is not `object`.